tests/libs/estdlib/test_supervisor.erl

```diff
@@ -340,3 +444,216 @@ test_which_children() ->
+        after 2000 ->
+            error({supervisor_not_stopped, reached_max_restart_intensity})
+        end,
+    process_flag(trap_exit, false),
+
+    %% Test child crashed and was not restarted
+    ok =
+        try gen_server:call(Pid5, ping) of
+            pong -> error(not_stopped, Pid5)
+        catch
+            exit:{noproc, _MFA} -> ok
+        end,
+    ok =
+        try get_ping_pong_pid() of
+            Pid6 when is_pid(Pid6) ->
+                error({child_restarted, reached_max_restart_intensity})
+        catch
+            throw:timeout ->
+                ok
+        end,
+
+    ok =
+        try erlang:process_info(SupPid, links) of
+            {links, Links} when is_list(Links) ->
+                error({not_stopped, reached_max_restart_intensity});
+            undefined ->
+                ok
+        catch
+            error:badarg ->
+                ok
+        end,
+    ok.
+
+get_ping_pong_pid() ->
+    receive
+        {ping_pong_server_ready, Pid} -> Pid
+    after 2000 -> throw(timeout)
+    end.
+
+try_again_restart() ->
+    process_flag(trap_exit, true),
+    timer:sleep(100),
+    Arbitrator = erlang:spawn(fun() -> arbitrator_start(4) end),
+    ok =
+        case is_pid(Arbitrator) of
+            true -> ok;
+            false -> {error, no_arbitrator}
+        end,
+    {ok, SupPid} = supervisor:start_link(
+        {local, try_again_test}, ?MODULE, {test_try_again, Arbitrator, self()}
+    ),
+    ChildPid =
+        case is_pid(SupPid) of
+            true ->
+                receive
+                    Pid0 when is_pid(Pid0) ->
+                        Pid0
+                after 200 ->
+                    error({timeout, no_child_start})
+                end;
+            false ->
+                {error, no_supervisor}
+        end,
+    ok =
+        case is_pid(ChildPid) of
+            true -> ok;
+            false -> ChildPid
+        end,
+    ChildPid ! ok,
+    ChildPid1 =
+        receive
+            Pid1 when is_pid(Pid1) ->
+                Pid1
+        after 2000 ->
+            error({timeout, no_child_restart})
+        end,
+    ChildPid1 ! ok,
+    Arbitrator ! shutdown,
+    exit(SupPid, normal),
+    ok =
+        receive
+            {'EXIT', SupPid, normal} ->
+                ok
+        after 2000 ->
+            error({supervisor_not_stopped, normal})
+        end,
+
+    Arbitrator2 = erlang:spawn(fun() -> arbitrator_start(5) end),
+    {ok, SupPid2} = supervisor:start_link(
+        {local, test_try_again}, ?MODULE, {test_try_again, Arbitrator2, self()}
+    ),
+    ChildPid2 =
+        case is_pid(SupPid2) of
+            true ->
+                receive
+                    Pid2 when is_pid(Pid2) ->
+                        Pid2
+                after 200 ->
+                    error({timeout, no_child2_start})
+                end;
+            false ->
+                {error, no_supervisor2}
+        end,
+    ok =
+        case is_pid(ChildPid) of
```

Comment:
believe you want ChildPid2 here

Reply:
Yes I do! Thank you. I reworked this function a bit too, I decided it was a waist of CI time to wait 2000 ms for a child message if the supervisor failed to start 🤦 . I moved the check for a supervisor pid sooner and a fixed the variable name.